src/lib/parseParams.ts

```diff
@@ -0,0 +1,200 @@
+import { cosmiconfig } from "cosmiconfig";
+import * as YAML from "yaml";
+import * as fs from "fs";
+
+type DenoifyParams = {
+    replacer?: string;
+    ports?: {
+        [portName: string]: string;
+    };
+    out?: string;
+    index?: string;
+    includes?: (
+        | string
+        | {
+              src: string;
+              destDir?: string;
+              destBasename?: string;
+          }
+    )[];
+};
+
+type ConfigFileType =
+    | {
+          type: "absent";
+      }
+    | {
+          type: "json" | "js" | "yaml" | "extensionless";
+          file: string;
+          config: string;
```

Comment:
What do you thing about renaming `file` to `configFileBasename` and `config` to `configFileRawContent` ?

Reply:
Sure, it convey better meaning